features/22-email-verification/api.md

```diff
@@ -0,0 +1,111 @@
+# Verify by email
+The use case and requirement apply to verify-by-email and verify-by-sms.
+
+## Use cases
+* App may request the user to verify the email address/phone number upon signup.
+* The app need will send an email/SMS to the user with the email address/phone number provided when logging in. The app will want to customize the content of the email/SMS.
+* Both email/SMS verification supports verify by a verification code or by a verification link.
+	* For verification code, the user will type the verification code from the email address/SMS an enter it to the app to complete verification.
+	* For verification link, the user will be redirected to a webpage, which automatically send the code to the server to complete verification.
+* The App is able to check the verification status of the user using the SDK and cloud function so that it can display other info and/or disable features.
+* The developer may want to deny unverified user from accessing any server API except auth related ones.
+
+## Requirements
+* Server will have configuration options for:
+	* enable/disable email verification (boolean)
+	* enable/disable SMS verification (boolean)
+	* set verification as required/optional (boolean)
```

Comment:
What will happen on API level if the verification is required / not required?

Reply:
I am thinking that if verification is required, then all the API (except for the auth related) will deny the request. We still want auth related API to go through because we still need the user to login so that they can ask the server to resend the verification email/SMS. (Otherwise, a user will be able to request a verification email for another user, which sounds bad : )

Reply:
Thanks understood. I guess a bit of explain in the spec will helps since it is not very trivial.